test/test_upload_processor.py

```diff
@@ -0,0 +1,159 @@
+            self.assertTrue(validate_move(self.upload_prefix, self.main_prefix, sample))
+
+    def test_batch_move_recovery(self):
+        """Test cases that handles previous partially successful run.
+        In this case, the file would not exist at the source
+        but would exist at the destination"""
+
+        # Rather than uploading files to the upload bucket, files are added
+        # To the main bucket, to mimic this behaviour
+        sample_list = ['Sample1.gVCF', 'Sample2.gVCF', 'Sample3.gVCF']
+        upload_files(sample_list, self.main_prefix)
+        recovery_batch = hb.Batch(name='Test Batch Move Recovery')
+        batch_move_files(
+            recovery_batch,
+            sample_list,
+            self.upload_prefix,
+            self.main_prefix,
+            self.docker_image,
+            self.key,
+        )
+
+        recovery_batch.run()
+
+        # Check that the files have been moved to main
+        for sample in sample_list:
+            self.assertTrue(validate_move(self.upload_prefix, self.main_prefix, sample))
+
+    def test_invalid_samples(self):
+        """Test case that handles invalid sample ID's i.e. samples that don't exist
+        in the upload bucket"""
+        sample_list = ['Sample8.gVCF', 'Sample9.gVCF']
+        invalid_batch = hb.Batch(name='Invalid Batch')
+        batch_move_files(
+            invalid_batch,
+            sample_list,
+            self.upload_prefix,
+            self.main_prefix,
+            self.docker_image,
+            self.key,
+        )
+
+        with self.assertRaises(subprocess.CalledProcessError):
+            invalid_batch.run()
+
+    def test_partial_recovery(self):
+        """Another partial recovery test case. In this scenario,
+        half the files were moved in a previous run and half were not."""
+
+        sample_list_failed = ['SampleA.gVCF', 'SampleB.gVCF']
+        sample_list_successful = ['SampleC.gVCF', 'SampleD.gVCF']
+
+        # Uploading this 'successful' files to the main bucket
+        upload_files(sample_list_successful, self.main_prefix)
+
+        # Uploading the 'failed' files to the upload bucket
+        # i.e. replicating the case were they have not moved.
+        upload_files(sample_list_failed, self.upload_prefix)
+
+        sample_list = sample_list_failed + sample_list_successful
+
+        partial_batch = hb.Batch(name='Test Batch Move Partial Recovery')
+        batch_move_files(
+            partial_batch,
+            sample_list,
+            self.upload_prefix,
+            self.main_prefix,
+            self.docker_image,
+            self.key,
+        )
+
+        partial_batch.run()
+
+        for sample in sample_list:
+            self.assertTrue(validate_move(self.upload_prefix, self.main_prefix, sample))
+
+    def tearDown(self):
+        # Deleting files created in test run
+        full_path = os.path.join('gs://', self.main_prefix, '*')
+        subprocess.run(['gsutil', 'rm', full_path], check=False)
```

Comment:
Do you want to add a `main` again, so this can be run easily in isolation?

Reply:
I usually run it with `-m unittest`. I added `main` previously for the analysis runner, which can now take in the additional arguments. Do you think it would be beneficial to add it back?

Reply:
I think it's nice to be able to make a change and _easily / quickly_ run the unit tests afterwards -- it makes it much more likely that people actually use / run / contribute to tests going forward.

Reply:
Great point, have added :)